pkg/scheduler/scheduler_test.go

```diff
@@ -1404,12 +1423,12 @@ func TestEntryAssignFlavors(t *testing.T) {
 				}),
 			}
 			tc.clusterQueue.UpdateLabelKeys(resourceFlavors)
-			fits := e.assignFlavors(log, resourceFlavors, &tc.clusterQueue)
-			if fits != tc.wantFits {
-				t.Errorf("e.assignFlavors(_)=%t, want %t", fits, tc.wantFits)
+			status := e.assignFlavors(log, resourceFlavors, &tc.clusterQueue)
+			if status.IsSuccess() != tc.wantFits {
+				t.Errorf("e.assignFlavors(_)=%t, want %t", status.IsSuccess(), tc.wantFits)
 			}
 			var flavors map[string]map[corev1.ResourceName]string
-			if fits {
+			if status.IsSuccess() {
```

Comment:
Should we test the message if it doesn't fit?

Reply:
Cover unfit message verifications.